docs/language/functions/grok.md

```diff
@@ -1,38 +1,159 @@
 ### Function
 
-&emsp; **grok** &mdash; parse a string using a grok pattern
+&emsp; **grok** &mdash; parse a string using a Grok pattern
 
 ### Synopsis
 
 ```
-grok(p: string, s: string) -> any
-grok(p: string, s: string, definitions: string) -> any
+grok(p: string, s: string) -> record
+grok(p: string, s: string, definitions: string) -> record
 ```
 
 ### Description
 
-The _grok_ function parses a string `s` using grok pattern `p` and returns
+The _grok_ function parses a string `s` using Grok pattern `p` and returns
 a record containing the parsed fields. The syntax for pattern `p`
 is `%{pattern:field_name}` where _pattern_ is the name of the pattern
 to match in `s` and _field_name_ is the resultant field name of the capture
 value.
 
 When provided with three arguments, `definitions` is a string
-of named patterns in the format `PATTERN_NAME PATTERN` each separated by newlines.
-The named patterns can then be referenced in argument `p`.
+of named patterns in the format `PATTERN_NAME PATTERN` each separated by
+newlines (`\n`). The named patterns can then be referenced in argument `p`.
 
-#### Included Patterns
+### Included Patterns
 
-The _grok_ function by default includes a set of builtin named patterns
+The `grok` function by default includes a set of built-in named patterns
 that can be referenced in any pattern. The included named patterns can be seen
 [here](https://raw.githubusercontent.com/brimdata/zed/main/pkg/grok/base.go).
 
+### Comparison to Other Implementations
```

Comment:
I know this section is long, but it's my best attempt to deal with the fact that web searches for "grok parsing" and such will reveal tons of examples that, taken unmodified, will not work in our implementation. Since we collectively don't have the cycles or stomach to build full compatibility with Logstash, I figure the next best thing is just disclose as much as we can and hope users are patient.